### PR TITLE
Add ADR for change in GOPATH

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Recently changed the location of `$GOPATH` and we need to update our ADRs about it.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -78,5 +78,6 @@ If you are looking to understand choices made in this project, see the list of [
 * 0033 [*Service Object Layer*](adr/0033-service-object-layer.md#service-object-layer)
 * 0034 [*Working With Mocks: Generation and Assertion*](adr/0034-working-with-mocks-generation-and-assertion.md#working-with-mocks-generation-and-assertion)
 * 0035 [Use Query Builder for for Admin Interface](adr/0035-use-query-builder.md#use-query-builder-for-for-admin-interface)
+* 0037 [Put mymove outside of standard GOPATH](adr/0037-go-path-and-project-layout-revisited.md#put-mymove-outside-of-standard-gopath)
```
Comment (chrisgilmerproj) [inline docs/README.md]: There's another PR for ADR0036.

File: docs/adr/0037-go-path-and-project-layout-revisited.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+**NOTE:** This ADR updates and supersedes [ADR0003 Go Path and Project Layout](./0003-go-path-and-project-layout.md).
+
+MilMove adopted using the go module system as introduction in `go1.12` in [PR 1932](https://github.com/transcom/mymove/pull/1932).
+This gave us the option of continuing to host our repository inside of `$GOPATH` or outside of `$GOPATH` and per
+[ADR0003 Go Path and Project Layout](./0003-go-path-and-project-layout.md) the decision was to stay in `$GOPATH` and
+set the environment variable `GO111MODULE=on` as a way of strictly enforcing go module behavior.
+
+This choice had implications for other tools that the project uses like `pre-commit` hooks that use golang libraries.  As seen in
+[PR 2236](https://github.com/transcom/mymove/pull/2236) the interaction made it impossible to develop the project and
+thus people `GO111MODULE=auto` was set and people were asked to move their repository outside of `$GOPATH`. This also
```
Comment (mikena-truss) [inline docs/adr/0037-go-path-and-project-layout-revisited.md]: "thus people set `GO111MODULE=auto` and were asked"

File: docs/adr/0037-go-path-and-project-layout-revisited.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+brings the local development inline with our CI/CD pipeline choices as modified in [PR 2172](https://github.com/transcom/mymove/pull/2172).
+
+## Considered Alternatives
+
+* Maintain direction of ADR0003 and keep repository checkout inside `$GOPATH` with `GO111MODULE=on`
+* Move repository outside of `$GOPATH` with `GO111MODULE=auto`
+
+## Decision Outcome
+
+* Move repository outside of `$GOPATH` and explicitly set `GO111MODULE=auto`.
```
Comment (mikena-truss) [inline docs/adr/0037-go-path-and-project-layout-revisited.md]: Could you give a brief definition of `auto` vs `strict`?

File: docs/adr/0037-go-path-and-project-layout-revisited.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Put mymove outside of standard GOPATH
+
+**NOTE:** This ADR updates and supersedes [ADR0003 Go Path and Project Layout](./0003-go-path-and-project-layout.md).
```
Comment (lynzt) [inline docs/adr/0037-go-path-and-project-layout-revisited.md]: do we want to add a deprecated warning in `ADR0003` and a link to this ADR?

File: docs/adr/0037-go-path-and-project-layout-revisited.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+MilMove adopted using the go module system as introduction in `go1.12` in [PR 1932](https://github.com/transcom/mymove/pull/1932).
+This gave us the option of continuing to host our repository inside of `$GOPATH` or outside of `$GOPATH` and per
+[ADR0003 Go Path and Project Layout](./0003-go-path-and-project-layout.md) the decision was to stay in `$GOPATH` and
+set the environment variable `GO111MODULE=on` as a way of strictly enforcing go module behavior.
```
Comment (lynzt) [inline docs/adr/0037-go-path-and-project-layout-revisited.md]: a link to more info about `GO111MODULE`? 
like here (if this is the best location for info): https://github.com/golang/go/wiki/Modules

File: docs/adr/0037-go-path-and-project-layout-revisited.md (hunk 4 above)
Comment (chrisgilmerproj) [reply docs/adr/0037-go-path-and-project-layout-revisited.md]: Yeah, good idea!

File: docs/adr/0037-go-path-and-project-layout-revisited.md (hunk 5 above)
Comment (chrisgilmerproj) [reply docs/adr/0037-go-path-and-project-layout-revisited.md]: Yep, will add it!

**Pull request merged**